src/DocGenerator.php

```diff
@@ -81,30 +81,27 @@ public function generateProjectTemplatesStructure(): void
         $this->parser->parse();
         $entitiesCollection = $this->rootEntityCollectionsGroup->get(ClassEntityCollection::NAME);
 
-        $openaiKey = getenv('OPENAI_API_KEY') ?: $this->io->askHidden('Enter the key to work with ChatGpt');
-        $openaiClient = \Tectalic\OpenAi\Manager::build(
-            new \GuzzleHttp\Client(),
-            new \Tectalic\OpenAi\Authentication($openaiKey)
-        );
-
-        $availableModels = array_values(array_filter(
-            array_map(
-                fn(array $v) => $v['id'],
-                $openaiClient->models()->list()->toArray()['data'] ?? []
-            ),
-            fn(string $v) => str_starts_with($v, "gpt-")
-        ));
-
-        $model = $this->io->choice("Choose GPT model from available", $availableModels);
-        $templatesStructureGenerator = new TemplatesStructureGenerator($openaiClient, $model);
+        $aiHandler = ProviderFactory::create();
```

Comment:
It seems to me that we should ask the user which provider them wants to use. Further, when starting work, if some data is missing, we should immediately ask about it, for example, when using openAI, we will ask for API_KEY and model type. Now we find out about these settings too late, when a person has already entered some specific data about the project. Or is there a second option, do not terminate the program with an exception, but ask for the missing data during operation